### PR TITLE
Consolidate strings

### PR DESCRIPTION
**Pull request by boek**
This PR does a handful of things:

1. It moves all strings to strings.swift.
2. It removes all duplicate `.strings` references in the project
3. It moves the one set of `.strings` references to Shared
4. Adds a new function that calls `NSLocalizedString` and replaces all strings to use this new function
5. Adds new localization function to Localized Strings Macro build setting

By doing all of this our import and export process should be much easier as we will no longer have to:
1. Cleanup exported xliffs as every string will only be defined once
2. When we import we no longer have to duplicate the `.strings` files to every target

---
image

Verified translations in widgets still work which means everything should be working :)

I tried to break this up into tiny commits to make it easier to review but it will still probably be worth reviewing together
┆Issue is synchronized with this [Jira Task](https://jira.mozilla.com/browse/FXIOS-1074)

### CODE REVIEW

File: Client.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -308,8 +303,6 @@
 		4F514FD41ACD8F2C0022D7EA /* HistoryTests.swift in Sources */ = {isa = PBXBuildFile; fileRef = 4F514FD31ACD8F2C0022D7EA /* HistoryTests.swift */; };
 		4F97573B1AFA6F37006ECC24 /* readerContent.html in Resources */ = {isa = PBXBuildFile; fileRef = 4F9757391AFA6F37006ECC24 /* readerContent.html */; };
 		5002717C41BC7C50F67F1CAD /* StateTests.swift in Sources */ = {isa = PBXBuildFile; fileRef = 50027345B49B967409DDA348 /* StateTests.swift */; };
-		52262904252505630083DFD5 /* CronTabsPerfTests.swift in Sources */ = {isa = PBXBuildFile; fileRef = 52262903252505630083DFD5 /* CronTabsPerfTests.swift */; };
```
Comment (dnarcese) [inline Client.xcodeproj/project.pbxproj:311]: Why is this test file removed?

File: Client.xcodeproj/project.pbxproj [hunk 2]
```diff
@@ -565,12 +550,27 @@
 		D3E171C21A841EAD00AB44CD /* KIFHelper.js in Resources */ = {isa = PBXBuildFile; fileRef = D3E171C11A841EAD00AB44CD /* KIFHelper.js */; };
 		D3E8EF101B97BE69001900FB /* ClearPrivateDataTableViewController.swift in Sources */ = {isa = PBXBuildFile; fileRef = D3E8EEE71B97A87A001900FB /* ClearPrivateDataTableViewController.swift */; };
 		D3FA777B1A43B2990010CD32 /* SearchTests.swift in Sources */ = {isa = PBXBuildFile; fileRef = D3FA777A1A43B2990010CD32 /* SearchTests.swift */; };
-		D3FB4CFA9F432DA3A605B399 /* 3DTouchActions.strings in Resources */ = {isa = PBXBuildFile; fileRef = 514A4C83AE17612AF3025498 /* 3DTouchActions.strings */; };
 		D3FEC38D1AC4B42F00494F45 /* AutocompleteTextField.swift in Sources */ = {isa = PBXBuildFile; fileRef = D3FEC38C1AC4B42F00494F45 /* AutocompleteTextField.swift */; };
-		D47A50F7254840210019A838 /* L10nSuite3SnapshotTests.swift in Sources */ = {isa = PBXBuildFile; fileRef = D47A50F6254840210019A838 /* L10nSuite3SnapshotTests.swift */; };
```
Comment (dnarcese) [inline Client.xcodeproj/project.pbxproj:570]: and this one

**Review by nbhasin2 — APPROVED** [review]

File: Client.xcodeproj/project.pbxproj [hunk 3]
```diff
@@ -7510,7 +7214,6 @@
 				7BEB64451C7345600092C02E /* SnapshotHelper.swift in Sources */,
 				D88C01B722B29EC200936951 /* ScreenGraphTest.swift in Sources */,
 				E40AFC541DD0E93300DA5651 /* L10nPermissionStringsSnapshotTests.swift in Sources */,
-				D47A50F7254840210019A838 /* L10nSuite3SnapshotTests.swift in Sources */,
```
Comment (nbhasin2) [inline Client.xcodeproj/project.pbxproj:7513]: L10nSuite3SnapshotTests as well

**Comment by nbhasin2** [comment]
Giving it a primarily approval apart from some files missing from pbjxproj (should be an easy fix / explanation)

**Comment by boek** [comment]
No idea how those files went missing! I looked through again and didn't see any more.. must have been during the rebase 😬

I also put up the l10n scripts up here. Going to document them tomorrow in the morning. https://github.com/boek/ios-l10n-scripts/tree/swiftify

**Comment by dnarcese** [comment]
Can you pull in this commit https://github.com/mozilla-mobile/firefox-ios/commit/7a2ce82de64da090641c92dd3c25184465190a45

**Pull request merged**